### PR TITLE
feat: ReCaptcha

### PR DESCRIPTION
**Pull request by aarbanas**
## Description

The main purpose of this PR was to introduce the ReCaptcha on the registration form, but we've made some code cleanup and extension as well. Everything is listed bellow. 

## Change summary
- ReCaptcha
- Reusable user profile picture component
- Toast to notify users of registration
- remove unused icons
- Fix bug in middleware redirection to 404
- Add `Admin button` on the user layout and `back to dashboard` on the admin layout just for admins

## Checklist

- [x] I have performed a self-review of my code
- [x] I tested locally
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New package or package update introduced

### CODE REVIEW

File: src/app/register/page.tsx [hunk 1]
```diff
@@ -2,45 +2,65 @@
...
   }, [watch]);
 
-  async function submit() {
+  const submit = async () => {
     setSubmitting(true);
+    if (!recaptchaRef?.current?.getValue()) {
+      toast('Molimo vas da potvrdite da niste robot', { type: 'error' });
+      setSubmitting(false);
+      return;
+    }
```
Comment (IMilja) [inline src/app/register/page.tsx:39]: I think there is no need to set the form submission as true until we start the submission of the form.

```suggestion
if (!recaptchaRef?.current?.getValue()) {
  toast('Molimo vas da potvrdite da niste robot', { type: 'error' });
  return;
}
```

File: src/app/register/page.tsx [hunk 2]
```diff
@@ -2,45 +2,65 @@
...
+
+    const recaptchaValue = recaptchaRef.current.getValue();
     const values = form.getValues();
 
+    setSubmitting(false);
+
     try {
       // eslint-disable-next-line @typescript-eslint/no-unused-vars
       const { repeatPassword, ...rest } = values;
```
Comment (IMilja) [inline src/app/register/page.tsx:48]: We can move the setSubmitting into the try statement.
```suggestion
    try {
      setSubmitting(true);
      // eslint-disable-next-line @typescript-eslint/no-unused-vars
      const { repeatPassword, ...rest } = values;
```

**Review by IMilja — COMMENTED** [review]
Left a few comments that we can look into.

File: src/app/register/page.tsx (hunk 1 above)
Comment (aarbanas) [reply src/app/register/page.tsx]: Yes, I fully agree!

File: src/app/register/page.tsx (hunk 2 above)
Comment (aarbanas) [reply src/app/register/page.tsx]: Yes, I fully agree!

**Review by IMilja — APPROVED** [review]
LGTM!

**Pull request merged**